sdk/nodejs/resource.ts

```diff
@@ -1349,9 +1410,34 @@ export function mergeOptions(opts1: ResourceOptions | undefined, opts2: Resource
             continue;
         }
 
+        if (key === "hooks") {
+            continue;
+        }
+
         dest[key] = merge(destVal, sourceVal, /*alwaysCreateArray:*/ false);
     }
 
+    if (dest.hooks || source.hooks) {
+        const hookTypes = [
+            "beforeCreate",
+            "afterCreate",
+            "beforeUpdate",
+            "afterUpdate",
+            "beforeDelete",
+            "afterDelete",
+        ] as const;
+        for (const hookType of hookTypes) {
+            const destHooks = dest?.hooks?.[hookType];
+            const sourceHooks = source?.hooks?.[hookType];
+            if (destHooks || sourceHooks) {
+                if (!dest.hooks) {
+                    dest.hooks = {};
+                }
+                dest.hooks[hookType] = merge(destHooks, sourceHooks, /*alwaysCreateArray:*/ false);
+            }
+        }
+    }
```

Comment:
Out of interest (not sure if worth a comment), why doesn't the default recursive merge work here? Does it not preserve ordering, or is there some other risk?